packages/subsecond/subsecond/src/lib.rs

```diff
@@ -284,6 +284,10 @@ pub struct HotFnPanic {
     _backtrace: backtrace::Backtrace,
 }
 
+/// A pointer to a hot patched function
+#[derive(PartialEq, Eq, Hash, Clone, Copy, Debug)]
+pub struct HotFnPtr(pub u64);
```

Comment:
```suggestion
#[non_exhaustive]
pub struct HotFnPtr(pub u64);

impl HotFnPtr {
    unsafe fn new(index: u64) -> Self {
        Self(index)
    }
}
```

Please reconsider using `#[non_exhaustive]` for the `struct`, I think that's a much better solution than resorting to making more API `unsafe`.